### DISCUSSION TITLE
torch.cuda.is_available() always return false

### DISCUSSION THREAD

**Discussion by knwin** [opengeos/segment-geospatial#115] [Q&A] [answered]
I have installed segment geospatial as instructed on the site. OS is windows 11 with Nvidia RTX 4050 (6GB).
It seems it does not use GPU as it always return False from torch.cuda.is_available(). 
How do I resolve it?

**Comment by giswqs** [comment, marked as answer]
Check out this [blog post](http://www.spatialpoints.com/2022/04/20/deep-learning-in-arcgis-pro-using-gpu/). 

A summary of the steps:
- Step 1: Install Microsoft Visual Studio
	- [Download](https://visualstudio.microsoft.com/downloads/) and install Microsoft visual studio community version. Select .NET desktop development and Desktop development with C++
- Step 2: Install Nvidia CUDA Toolkit
	- http://developer.nvidia.com/cuda-downloads
- Test cuda with nvidia-smi

**Reply by knwin** [reply]
Thanks Sir,
I have failed after many tries.  Torch modules is always cpu version when install via mamba/conda.
So I make another installation with pip. I additionally installed cuda verson of torch as follow.. (ref: [here](https://stackoverflow.com/questions/57814535/assertionerror-torch-not-compiled-with-cuda-enabled-in-spite-upgrading-to-cud/70509457#70509457))
pip3 install --ignore-installed torch==2.0.1 torchvision==0.15.2 --index-url https://download.pytorch.org/whl/cu117 

Then torch.cuda.is_available() return true.

( I ran into another problem while running sample note book :( )

Thanks for your support.

**Comment by oscarbau** [comment]
I faced the same issue, you need to be really careful with the versions of Microsoft Visual Studio, Nvidia CUDA Toolkit, GPU model, and Python in your environment. They all need to be compatible with each other

**Reply by knwin** [reply]
I found multiple Microsoft Visual Studio in my laptop. 2010,2012 (x64 & x86), 2013 (x64 & x86), 2015-2022 (x64 & x86). Is it ok to have multiple versions?

**Reply by oscarbau** [reply]
I don't know, you need to test.
